projectq/types/_qubit.py

```diff
@@ -116,12 +116,18 @@ class Qubit(BasicQubit):
     Thus the qubit is not copyable; only returns a reference to the same
     object.
     """
+    def __init__(self, engine, idx):
+        BasicQubit.__init__(self, engine, idx)
+
     def __del__(self):
         """
         Destroy the qubit and deallocate it (automatically).
         """
-        self.engine.deallocate_qubit(self)
+        if self.id == -1:
```

Comment:
Done. `deallocate_qubit` now asserts the qubit id isn't -1 as a precondition, and `__del__` satisfies that precondition by avoiding the call. People who have a bug that accidentally frees qubit twice will be appropriately warned by the exception.